tests/conftest.py

```diff
@@ -0,0 +1,198 @@
+import json
+from typing import Tuple
+
+import pytest
+import requests
+from _pytest.fixtures import FixtureRequest
+from kinto_http import Client
+from requests.adapters import HTTPAdapter
+from selenium.webdriver.firefox.options import Options
+from selenium.webdriver.remote.webdriver import WebDriver
+from urllib3.util.retry import Retry
+
+
+DEFAULT_SERVER = "http://localhost:8888/v1"
+DEFAULT_AUTH = "user:pass"
+DEFAULT_EDITOR_AUTH = "editor:pass"
+DEFAULT_REVIEWER_AUTH = "reviewer:pass"
+DEFAULT_BUCKET = "source"
+DEFAULT_COLLECTION = "source"
+
+
+def pytest_addoption(parser):
+    parser.addoption(
+        "--server",
+        action="store",
+        default=DEFAULT_SERVER,
+        help="Kinto server (in form 'http(s)://<host>:<port>/v1')",
+    )
+    parser.addoption(
+        "--auth",
+        action="store",
+        default=DEFAULT_AUTH,
+        help="Basic authentication",
+    )
+    parser.addoption(
+        "--editor-auth",
+        action="store",
+        default=DEFAULT_EDITOR_AUTH,
+        help="Basic authentication for editor",
+    )
+    parser.addoption(
+        "--reviewer-auth",
+        action="store",
+        default=DEFAULT_REVIEWER_AUTH,
+        help="Basic authentication for reviewer",
+    )
+    parser.addoption(
+        "--bucket",
+        action="store",
+        default=DEFAULT_BUCKET,
+        help="Source bucket",
+    )
+    parser.addoption(
+        "--collection",
+        action="store",
+        default=DEFAULT_COLLECTION,
+        help="Source collection",
+    )
+    parser.addoption(
+        "--reset",
+        action="store_true",
+        default=False,
+        help="Reset collection data",
+    )
+
+
+@pytest.fixture(scope="session")
+def server(request):
+    return request.config.getoption("--server")
+
+
+@pytest.fixture(scope="session")
+def auth(request):
+    return tuple(request.config.getoption("--auth").split(":"))
+
+
+@pytest.fixture(scope="session")
+def editor_auth(request):
+    return tuple(request.config.getoption("--editor-auth").split(":"))
+
+
+@pytest.fixture(scope="session")
+def reviewer_auth(request):
+    return tuple(request.config.getoption("--reviewer-auth").split(":"))
+
+
+@pytest.fixture(scope="session")
+def source_bucket(request):
+    return request.config.getoption("--bucket")
+
+
+@pytest.fixture(scope="session")
+def source_collection(request):
+    return request.config.getoption("--collection")
+
+
+@pytest.fixture(scope="session")
+def reset(request):
+    return request.config.getoption("--reset")
+
+
+@pytest.fixture
+def get_clients(
```

Comment:
Do you think it would be good to split this into separate fixtures, or at least not have this fixture generate three clients by default? If I read the test cases correctly, I think there's only one test case where we need more than the User client. Also, creating all three users each time we need just one client seems like an unexpected side effect.

Maybe we could use the [Factory as fixture](https://docs.pytest.org/en/latest/how-to/fixtures.html#factories-as-fixtures) pattern here. So the fixture would be something like 
```python
@pytest.fixture
def make_client():
    def _make_client(auth):
        ...
    return _make_client
```
then in tests, we can do:
```python
def test_with_client(make_client, auth):
    client = make_client(auth)
    ...

def test_with_clients(make_client, auth, editor_auth, reviewer_auth):
    client = make_client(auth)
    editor_client = make_client(editor_auth)
    reviewer_client = make_client(reviewer_auth)
    ...
```

Reply:
This is awesome, Graham! Thanks for the great idea

Reply:
https://github.com/mozilla-services/kinto-dist/pull/1953/commits/d9cb5bf0984244f69fb09bb42a9eb6f8846f7615